extern/storage-sealing/commit_batch.go

```diff
@@ -452,24 +456,36 @@ func (b *CommitBatcher) Stop(ctx context.Context) error {
 	}
 }
 
-func getSectorDeadline(curEpoch abi.ChainEpoch, si SectorInfo) time.Time {
-	deadlineEpoch := si.TicketEpoch + policy.MaxPreCommitRandomnessLookback
+func (b *CommitBatcher) getCommitCutoff(si SectorInfo) (time.Time, error) {
+	tok, curEpoch, err := b.api.ChainHead(b.mctx)
+	if err != nil {
+		log.Errorf("getting chain head: %s", err)
+		return time.Now(), err
+	}
+
+	nv, err := b.api.StateNetworkVersion(b.mctx, tok)
+	if err != nil {
+		log.Errorf("getting network version: %s", err)
+		return time.Now(), err
+	}
+
+	cutoffEpoch := si.PreCommitHeight + policy.GetMaxProveCommitDuration(actors.VersionForNetwork(nv), si.SectorType)
```

Comment:
If we've just added the PreCommitHeight field, it won't be present on existing sectors in existing miners